File: CellsToServers/HierarchicalTilingApp/ArrayPartition/Coords.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HierarchicalTilingApp.ArrayPartition
{
    public class Coords
    {
        public int[] ExtendedIndicesArray { get; set; }
        public int HeftOfRegion { get; set; }

        public double differenceFromDelta(double delta)
        {
            return Math.Abs(delta - HeftOfRegion);
        }

        public void printCoords(int spaceDimension, int serialNO)
        {
            Console.Write("{0}. tile: [", serialNO);
            for (int idx = 0; idx < spaceDimension; idx++)
            {
                Console.Write(" {0} {1} ", ExtendedIndicesArray[2 * idx + 1], ExtendedIndicesArray[2 * idx + 2]);
            }
            Console.WriteLine("] : {0} heft", HeftOfRegion);
        }

        public void writeToStringBuilder(int spaceDimension, StringBuilder strBldr)
        {
            strBldr.Append(HeftOfRegion);
            for (int idx = 0; idx < spaceDimension; idx++)
            {
                strBldr.Append(" ").Append(ExtendedIndicesArray[2 * idx + 1]).Append(" ").Append(ExtendedIndicesArray[2 * idx + 2]);
            }
            strBldr.AppendLine();
        }
    }
}

File: CellsToServersCpp/HierarchicalTilingAppCpp/ArrayPartition/Coords.cpp
#include "Coords.hpp"

namespace ArrayPartition
{
	int * Coords::getExtendedIndicesArray()
	{
		return extendedIndicesArray;
	}

	void Coords::setExtendedIndicesArray(int * inputExtendedIndicesArray)
	{
		this->extendedIndicesArray = inputExtendedIndicesArray;
	}

	int Coords::getHeftOfRegion()
	{
		return heftOfRegion;
	}

	void Coords::setHeftOfRegion(int inputHeftOfRegion)
	{
		this->heftOfRegion = inputHeftOfRegion;
	}

	double Coords::differenceFromDelta(double delta)
	{
		return abs(delta - heftOfRegion);
	}

	void Coords::printCoords(int spaceDimension, int serialNO)
	{
		std::cout << serialNO << ". tile: [";
		for (int idx = 0; idx < spaceDimension; idx++)
		{
			std::cout << " " << extendedIndicesArray[2 * idx + 1] << " " 
				<< extendedIndicesArray[2 * idx + 2] << " ";
		}
		std::cout << "] : " << heftOfRegion << " heft" << std::endl;
	}

	void Coords::writeToStringBuilder(int spaceDimension, std::stringstream& strBldr)
	{
		strBldr << heftOfRegion;
		for (int idx = 0; idx < spaceDimension; idx++)
		{
			strBldr << " " << extendedIndicesArray[2 * idx + 1] << " "
				<< extendedIndicesArray[2 * idx + 2];
		}
		strBldr << std::endl;
		//std::cout << strBldr.str();
	}
}